### PR TITLE
Discard truncate fn/v2

### PR DESCRIPTION
**Pull request by inashivb**
Link to ticket: https://redmine.openinfosecfoundation.org/issues/7044

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1953

Previous PR: https://github.com/OISF/suricata/pull/11393

Changes since v1:
- added explanation in the commit message
- removed all truncate glue code and parser calls

TODO: `smb` glue code is removed but the truncate fn still exists as it is used at several places. Nothing was affected on tests locally but would like to see if it has an impact in the bigger QA.

### CODE REVIEW

File: rust/src/smb/smb.rs [hunk 1]
```diff
@@ -2327,7 +2327,6 @@ pub unsafe extern "C" fn rs_smb_register_parser() {
         get_state_data: rs_smb_get_state_data,
         apply_tx_config: None,
         flags: APP_LAYER_PARSER_OPT_ACCEPT_GAPS,
-        truncate: Some(rs_smb_state_truncate),
```
Comment (victorjulien) [inline rust/src/smb/smb.rs:2330]: `rs_smb_state_truncate` seems otherwise uncalled?

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 21357

File: rust/src/smb/smb.rs (hunk 1 above)
Comment (inashivb) [reply rust/src/smb/smb.rs:2330]: Are you asking if it's uncalled in `master`?

Comment (victorjulien) [reply rust/src/smb/smb.rs:2330]: I see that it is, so why keep it?

File: src/app-layer-parser.c [hunk 2]
```diff
@@ -1468,9 +1462,9 @@ int AppLayerParserParse(ThreadVars *tv, AppLayerParserThreadCtx *alp_tctx, Flow
         AppLayerIncTxCounter(tv, f, cur_tx_cnt - p_tx_cnt);
     }
 
-    /* stream truncated, inform app layer */
+    /* stream truncated, trigger raw stream reassembly */
     if (flags & STREAM_DEPTH)
-        AppLayerParserStreamTruncated(pstate, f->proto, alproto, f->alstate, flags);
+        AppLayerParserTriggerRawStreamReassembly(f, direction);
```
Comment (catenacyber) [inline src/app-layer-parser.c:1455]: This looks strange to me, do we set another flag somewhere else when depth is reached ?

**Comment by catenacyber** [comment]
Commit removing Truncate looks good to me (besides need to remove `rs_smb_state_truncate` in addition)

Other commit looks suspicious... I guess it is ok but it needs more explanations...

File: src/app-layer-parser.c (hunk 2 above)
Comment (inashivb) [reply src/app-layer-parser.c:1455]: Q: flag for which part of the code exactly? applayer?

My idea here was that whenever we encounter stream depth condition, we make whatever data is available until now in the raw stream to the applayer. This is what was done in truncate fn calls too which I have removed.

Comment (victorjulien) [reply src/app-layer-parser.c:1455]: Is this not already done by the detect/stream engine understanding the depth logic?

Comment (catenacyber) [reply src/app-layer-parser.c:1455]: Why are the `APP_LAYER_PARSER_TRUNC_TS` flags removed ?
Are some other flags equivalent ?

`ts_disrupt_flags` gets `STREAM_DEPTH` flag from `FlowGetDisruptionFlags` which gets used by `AppLayerParserGetStateProgress` with `IS_DISRUPTED`
No need for APP_LAYER_PARSER_TRUNC_TS flag.
That is what I want to see somehow in the commit message

**Pull request closed without merging**